internal/activities/verify_manifest.go

```diff
@@ -199,6 +202,54 @@ func verifyChecksums(
 					),
 				)
 			}
+		case "SHA-1":
+			hash, err := sha1Hash(filepath.Join(root, path))
+			if err != nil {
+				return nil, fmt.Errorf("generate SHA-1 hash: %v", err)
+			}
+			if hash != file.Checksum.Hash {
+				failures = append(
+					failures,
+					fmt.Sprintf(
+						"Checksum mismatch for %q (expected: %q, got: %q)",
+						path,
+						file.Checksum.Hash,
+						hash,
+					),
+				)
+			}
+		case "SHA-256":
+			hash, err := sha256Hash(filepath.Join(root, path))
+			if err != nil {
+				return nil, fmt.Errorf("generate SHA-256 hash: %v", err)
+			}
+			if hash != file.Checksum.Hash {
+				failures = append(
+					failures,
+					fmt.Sprintf(
+						"Checksum mismatch for %q (expected: %q, got: %q)",
+						path,
+						file.Checksum.Hash,
+						hash,
+					),
+				)
+			}
+		case "SHA-512":
+			hash, err := sha512Hash(filepath.Join(root, path))
+			if err != nil {
+				return nil, fmt.Errorf("generate SHA-512 hash: %v", err)
+			}
+			if hash != file.Checksum.Hash {
+				failures = append(
+					failures,
+					fmt.Sprintf(
+						"Checksum mismatch for %q (expected: %q, got: %q)",
+						path,
+						file.Checksum.Hash,
+						hash,
+					),
+				)
+			}
```

Comment:
Early feedback, in case you didn't think about it already. Putting the switch in a single `generateHash` function reduces duplication:

```go
// Generate checksum from filesystem file contents.
hash, err := generateHash(filepath.Join(root, path), file.Checksum.Algorithm)
if err != nil {
	return nil, fmt.Errorf("generate %s hash: %v", file.Checksum.Algorithm, err)
}
if hash != file.Checksum.Hash {
	failures = append(
		failures,
		fmt.Sprintf(
			"Checksum mismatch for %q (expected: %q, got: %q)",
			path,
			file.Checksum.Hash,
			hash,
		),
	)
}
```

```go
func generateHash(path, alg string) (string, error) {
	var h hash.Hash
	switch alg {
	case "MD5":
		h = md5.New() // #nosec: G401 -- not used for security.
	case "SHA-1":
		h = sha1.New() // #nosec: G401 -- not used for security.
	case "SHA-256":
		h = sha256.New() // #nosec: G401 -- not used for security.
	case "SHA-512":
		h = sha512.New() // #nosec: G401 -- not used for security.
	default:
		return "", fmt.Errorf("hash algorithm %q is not supported", alg)
	}

	f, err := os.Open(path) // #nosec: G304 -- trusted path.
	if err != nil {
		return "", fmt.Errorf("open file: %v", err)
	}
	defer f.Close()

	if _, err := io.Copy(h, f); err != nil {
		return "", fmt.Errorf("copy contents: %v", err)
	}

	return hex.EncodeToString(h.Sum(nil)), nil
}
```